src/app/info/WikiItems.test.tsx

```diff
@@ -0,0 +1,281 @@
+                    <WikiItems rows={mockData} />
+                </RoleUpdateContext.Provider>
+            </StyleManager>
+        );
+        await user.click(screen.getByTestId("#edit-2"));
+        await user.click(screen.getByTestId("#swap-up-2"));
+        expect(mockData[0].row_order).toBe(2);
+        expect(mockData[1].row_order).toBe(1);
+    });
+
+    it("deletes a wiki item when its delete button is clicked", async () => {
+        const user = userEvent.setup();
+
+        render(
+            <StyleManager>
+                <RoleUpdateContext.Provider value={{ role: "admin", setRole: jest.fn() }}>
+                    <WikiItems rows={mockData} />
+                </RoleUpdateContext.Provider>
+            </StyleManager>
+        );
+        await user.click(screen.getByTestId("#edit-1"));
+        await user.click(screen.getByTestId("#delete-1"));
+        expect(screen.queryByText("Test 1")).toBeNull();
+    });
+
+    it("edits a wiki item when its edit button is clicked", async () => {
+        const user = userEvent.setup();
+
+        render(
+            <StyleManager>
+                <RoleUpdateContext.Provider value={{ role: "admin", setRole: jest.fn() }}>
+                    <WikiItems rows={mockData} />
+                </RoleUpdateContext.Provider>
+            </StyleManager>
+        );
+        await user.click(screen.getByTestId("#edit-1"));
+        const titleInput = screen.getByTestId("#title-1");
+        fireEvent.change(titleInput, { target: { value: "Updated Test 1" } });
+        const contentInput = screen.getByTestId("#content-1");
+        fireEvent.change(contentInput, { target: { value: "Updated Test content 1" } });
+        await user.click(screen.getByTestId("#update-1"));
+        expect(screen.queryByText("Test 1")).toBeNull();
+        expect(screen.getByText("Updated Test 1")).toBeInTheDocument();
+        expect(screen.queryByText("Test content 1")).toBeNull();
+        expect(screen.getByText("Updated Test content 1")).toBeInTheDocument();
+    });
+
+    it("adds an item when the add button is clicked", async () => {
+        const user = userEvent.setup();
+
+        render(
+            <StyleManager>
+                <RoleUpdateContext.Provider value={{ role: "admin", setRole: jest.fn() }}>
+                    <WikiItems rows={mockData} />
+                </RoleUpdateContext.Provider>
+            </StyleManager>
+        );
+        await user.click(screen.getByTestId("#add"));
+        const titleInput = screen.getByTestId("#title-3");
+        fireEvent.change(titleInput, { target: { value: "Added test title" } });
+        const contentInput = screen.getByTestId("#content-3");
+        fireEvent.change(contentInput, { target: { value: "Added test content" } });
+        await user.click(screen.getByTestId("#update-3"));
+        expect(screen.getByText("Added test title")).toBeInTheDocument();
+        expect(screen.getByText("Added test content")).toBeInTheDocument();
```

Comment:
This may sound strange but could we add coverage to ensure that no other items change/are added when the add button is clicked. The Info page used to have a habit of doing weird item display duplication so I think its probably a good idea